Add new PR templates and revamp issue templates (#3711)
EssentialsX/Essentials: .github/ISSUE_TEMPLATE/help.md

---

<!-- EssentialsX help request guide

NOTE: Failure to fill out this template properly may result in your issue being
      delayed or ignored.

Don't put anything inside this block, as it will be hidden when you post your

> PR. Make sure to also not type between any arrows in the template, as this text

Reviewer: `
```suggestion
issue. Make sure to also not type between any arrows in the template, as this text
```

> Don't put anything inside this block, as it will be hidden when you post your
> issue. Make sure to also not type between any arrows in the template, as this
> text will also be hidden.

Reviewer 2: Just a thought, but perhaps clarify what "this block" means, since I'd expect most of the people putting text inside it don't necessarily know what that means. Something like this, perhaps?
```suggestion
Don't type anything until the "Information" subsection below, as it will be
otherwise be hidden when you post your issue. Also make sure not to write
anything between any arrows in the template, as this text will also be hidden.
```

> Want faster support? Come join our Discord server to get help from both
> developers and community members: https://discord.gg/F7gexAQ

Reviewer 2: Oftentimes Discord support is not fast since it is entirely driven by volunteers most hours of the day, so it would be good not to get people's expectations up.
```suggestion
Want an answer sooner? Join our Discord server to get help from community
members and sometimes even the developers themselves! Simply ask and wait
to see if anyone is able to help: https://discord.gg/F7gexAQ
```

Author: The text in this change feels clunky, and it carries pretty much the same meaning as before?

Reviewer 2: Tried to keep most of the meaning the same, "sooner" does not imply "fast", and there is an emphasis put on community support over developer support (so people aren't expecting developers to always be available to respond). Lastly, "ask and wait" again to indicate that it may not be instant.

Could use some better wording, this was merely a suggestion.

If you're happy to wait (or you were sent here from Discord), read on:

1.  Check the Common Issues page.
      Read through the wiki page to see if you've encountered a regular issue:
      https://essentialsx.net/wiki/Common-Issues.html

2.  Fill out the template.
      Run the commands in the console. Don't just put "latest" as a version, or
      we will ignore it. This will help us understand what problem you've
      encountered and help us find a solution.

3.  When linking logs or config files, do not attach them to the post!
      Copy and paste any logs into https://gist.github.com/, then paste a
      link to them in the relevant parts of the template. Do not use Hastebin
      or Pastebin, as this can cause issues when trying to solve issues in the
      future.

>       DO NOT drag logs into this box!

Reviewer 2: Might not be needed with the clarification suggestion above.

Author: This is about attaching logs to GitHub, not about where they go in the template. Other comments clarify where to put them.

4.  If you are encountering a performance issue, please include a link to a

>     Timings and/or profiler report, both before and after your PR.

Reviewer 2: Pretty sure this isn't a PR template so we can remove this part.
```suggestion
    Timings and/or profiler report.
```

-->

### Information

**Full output of `/ess version`:**
<!--
    Paste the output of `/ess version` inside the ```s, preferably copied
    directly from the console.
-->
```

```

**Server log:**
<!--

>     Upload `logs/latest.log` to https://gist.github.com/ then paste the link
>     below this block.

Reviewer 2: This section of the template has always been under hot debate by people worried about privacy. Perhaps good to give them some peace of mind; this shouldn't affect us too much and we can always remind them to re-submit if they fail to include critical information.
```suggestion
    Upload `logs/latest.log` to https://gist.github.com/ then paste the link
    below this block. You may remove any sensitive information, but please
    make it clear what was removed.
```
